src/chocolatey/infrastructure.app/validations/SystemStateValidation.cs

```diff
@@ -0,0 +1,92 @@
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace chocolatey.infrastructure.app.validations
+{
+    using System;
+    using System.Collections.Generic;
+    using configuration;
+    using infrastructure.validations;
+    using services;
+
+    /// <summary>
+    ///   Performs validation against the current System State.  This
+    ///   includes things like pending reboot requirement.  Any errors
+    ///   that are returned will halt the current operation.
+    /// </summary>
+    public class SystemStateValidation : IValidation
+    {
+        private readonly IPendingRebootService _pendingRebootService;
+
+        public SystemStateValidation(IPendingRebootService pendingRebootService)
+        {
+            _pendingRebootService = pendingRebootService;
+        }
+
+        public ICollection<ValidationResult> validate(ChocolateyConfiguration config)
+        {
+            var validationResults = new List<ValidationResult>();
+
+            validationResults.Add(check_system_pending_reboot(config));
+
+            return validationResults;
+        }
+
+        private ValidationResult check_system_pending_reboot(ChocolateyConfiguration config)
+        {
+            var result = _pendingRebootService.is_pending_reboot(config);
+
+            if (result)
+            {
+                if (!config.Features.ExitOnRebootDetected)
+                {
+                    return new ValidationResult
+                    {
+                        Message = @"A pending system reboot request has been detected, however, this is
+being ignored due to the current Chocolatey configuration.  If you want
+to halt when this occurs, then either set the global feature using:
+
+choco feature enable -name={0}
+
+or, pass the parameter --exit-when-reboot-detected.".format_with(ApplicationParameters.Features.ExitOnRebootDetected),
+                        Status = ValidationStatus.Warning,
+                        ExitCode = 0
+                    };
+                }
+                else
+                {
+                    Environment.ExitCode = 350;
+
+                    return new ValidationResult
+                    {
+                        Message = "A pending system reboot has been detected.",
+                        Status = ValidationStatus.Error,
+                        ExitCode = 350
+                    };
+                }
+            }
+            else
+            {
+                return new ValidationResult
+                {
+                    Message = "System State is valid.",
+                    Status = ValidationStatus.Success,
+                    ExitCode = 0

```

Comment:
Same idea here - loop over the results after the checks in the `validate()` method and add this if there are no warnings/errors.

Reply:
In the end, I didn't enumerate, but rather simply count the results.  If there are 0, add the success message.